src/pipert2/core/managers/networks/queue_network.py

```diff
@@ -39,11 +39,15 @@ def link(self, source: Routine, destinations: Tuple[Routine], data_transmitter:
             data_transmitter: The data transmitter that indicates how to transfer the data.
 
         """
+
         publish_queue = PublishQueue()
 
         for destination_routine in destinations:
-            destination_routine.message_handler.input_queue = \
-                publish_queue.register(destination_routine.message_handler.input_queue)
+            if source.flow_name == destination_routine.flow_name:
+                publish_queue.register(destination_routine.message_handler.input_queue.get_queue(process_safe=False))
+            else:
+                publish_queue.register(destination_routine.message_handler.input_queue.get_queue(process_safe=True))
+
```

Comment:
Actually when looking at it, can be refactored to one line where the 
`source.flow_name == destination_routine.flow_name` condition can be used in the process_safe parameter.
But in the current version of code it is more readable, up to you if u want to keep it.